controller/controlplane/manager_options.go

```diff
@@ -93,3 +96,109 @@ func WithMetricsServerOff() managercfg.Opt {
 		c.MetricsAddr = "0" // 0 disables metrics server
 	}
 }
+
+// WithFeatureGates sets the feature gates for the manager.
+func WithFeatureGates(logger logr.Logger, featureGates []gwtypes.ControlPlaneFeatureGate) managercfg.Opt {
+	return func(c *managercfg.Config) {
+		fgs := managercfg.FeatureGates{}
+		defaults := managercfg.GetFeatureGatesDefaults()
+		for _, feature := range featureGates {
+			if _, ok := defaults[feature.Name]; !ok {
+				log.Error(logger, errors.New("unknown feature gate"), "unknown feature gate",
+					"feature", feature.Name, "state", feature.State,
+				)
+				continue
+			}
+
+			v := false
+			if feature.State == gwtypes.FeatureGateStateEnabled {
+				v = true
+			}
+			// This should never happen as it should be enforced at the CRD level
+			// but we handle it gracefully here and log an error.
+			if _, ok := fgs[feature.Name]; ok {
+				log.Error(logger, errors.New("feature gate already set"), "feature gate already set",
+					"feature", feature.Name, "state", feature.State,
+				)
+				continue
+			}
+			fgs[feature.Name] = v
```

Comment:
```suggestion
			fgs[feature.Name] = (feature.State == gwtypes.FeatureGateStateEnabled)
```
maybe just like that and get rid of lines `113:116`. WDYT?

Reply:
👍 PTAL aab6f12098419ba3222eebfcf6df0a88b9fa8434